src/megolm/inbound_group_session.rs

```diff
@@ -102,6 +119,58 @@ impl InboundGroupSession {
         base64_encode(self.signing_key.as_bytes())
     }
 
+    /// Check if two `InboundGroupSession`s are the same.
+    ///
+    /// An `InboundGroupSession` could be received multiple times with varying
+    /// degreess of trust and first known message indices.
+    ///
+    /// This method allows to check if both `InboundGroupSession`s at the same
+    /// message index provide the same message keys, i.e. if they have been
+    /// indeed created from the same [`GroupSession`].
```

Comment:
```suggestion
    /// This method checks if the underlying ratchets of the two
    /// `InboundGroupSession`s are actually the same ratchet, potentially at
    /// a different ratcheting index. That is, if the sessions are *connected*,
    /// then ratcheting one of the ratchets to the index of the other should
    /// yield the same ratchet value, byte-for-byte. This will only be the case
    /// if the `InboundGroupSession`s were created from the same
    /// [`GroupSession`].
```